yt_dlp/extractor/hidive.py

```diff
@@ -46,8 +46,24 @@ def _perform_login(self, username, password):
             'Email': username,
             'Password': password,
         })
-        self._download_webpage(
+        loginWebpage = self._download_webpage(
             self._LOGIN_URL, None, 'Logging in', data=urlencode_postdata(data))
+        # If the user has multiple profiles, select one.
+        profileIds = re.findall(r'\<button .+data-profile-id="([a-zA-Z0-9]+)".*\>', loginWebpage)
+        if profileIds is None:
+            return # If only one profile, Hidive auto-selects it.
+        profileIdHashes = re.findall(r'\<button .+data-hash="([a-zA-Z0-9]+)".*\>', loginWebpage)
+        # Just pick the first profile, until someone else updates this code to
+        #   support declaring a profile in a config file.
+        profileId = profileIds[0]
+        profileIdHash = profileIdHashes[0]
```

Comment:
* Use `self._search_regex`
* Simplify regex (`a-zA-Z0-9` => `\w` etc)
* Avoid `.*`

Reply:
Thanks! I didn't know there was a preference for that function! I've changed to use that one.